examples/erc1155-metadata-uri/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg_attr(not(test), no_main)]
+extern crate alloc;
+
+use alloy_primitives::{FixedBytes, U256};
+use openzeppelin_stylus::{
+    token::erc1155::{
+        extensions::{Erc1155MetadataUri, Erc1155UriStorage},
+        Erc1155,
+    },
+    utils::introspection::erc165::IErc165,
+};
+use stylus_sdk::prelude::{entrypoint, public, sol_storage};
+
+sol_storage! {
+    #[entrypoint]
+    struct Erc1155MetadataUriExample {
+        #[borrow]
+        Erc1155 erc1155;
+        #[borrow]
+        Erc1155MetadataUri metadata_uri;
+        Erc1155UriStorage uri_storage;
+    }
+}
+
+#[public]
+#[inherit(Erc1155, Erc1155MetadataUri)]
```

Comment:
IMHO you do not need to inherit `Erc1155MetadataUri` as you overwrote `uri`.

Reply:
@qalisander @bidzyyys does it make sense to do `impl IErc1155MetadataUri for Erc1155UriStorage`? :thinking: 
It does technically extend (and overwrite) its `uri` function

Reply:
p.s. @0xNeshi also we don't need `#[borrow]` for `Erc1155MetadataUri` in this case